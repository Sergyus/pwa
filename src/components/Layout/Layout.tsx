import Header from '../Header';
import Footer from '../Footer';
import React from 'react';
import { Container } from '@material-ui/core';
import { ReactNode } from 'react';

type Props = {
  children: NonNullable<ReactNode>;
};

export default function Layout(props: Props): JSX.Element {
  return (
    <>
      <div className="app-content">
        <Header />
        <Container maxWidth="lg">{props.children}</Container>
      </div>
      <Footer />
    </>
  );
}
