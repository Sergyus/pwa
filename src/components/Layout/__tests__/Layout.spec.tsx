import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Layout from '../Layout';

describe('<Layout />', () => {
  it('should render with children', () => {
    const CONTENT = 'content';
    const { asFragment } = render(
      <MemoryRouter>
        <Layout>
          <div>{CONTENT}</div>
        </Layout>
      </MemoryRouter>,
    );
    expect(screen.getByText(CONTENT)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
