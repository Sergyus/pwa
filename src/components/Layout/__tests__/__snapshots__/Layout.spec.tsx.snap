// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render with children 1`] = `
<DocumentFragment>
  <div
    class="app-content"
  >
    <div
      class="makeStyles-grow-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-noWrap"
          >
            <a
              href="/"
              style="color: rgb(255, 255, 255); text-decoration: none;"
            >
              Logo
            </a>
          </h6>
          <div
            class="makeStyles-sectionDesktop-8"
          >
            <div
              class="nav-header"
              data-testid="menu"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <a
                    data-testid="menuItem"
                    href="/"
                    style="color: rgb(255, 255, 255); text-decoration: none;"
                  >
                    Home
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <a
                    data-testid="menuItem"
                    href="/blog"
                    style="color: rgb(255, 255, 255); text-decoration: none;"
                  >
                    Blog
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <a
                    data-testid="menuItem"
                    href="/about"
                    style="color: rgb(255, 255, 255); text-decoration: none;"
                  >
                    About
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <a
                    data-testid="menuItem"
                    href="/ui"
                    style="color: rgb(255, 255, 255); text-decoration: none;"
                  >
                    Ui
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <a
                    data-testid="menuItem"
                    href="/pwa"
                    style="color: rgb(255, 255, 255); text-decoration: none;"
                  >
                    Pwa
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <a
                    data-testid="menuItem"
                    href="/apollo"
                    style="color: rgb(255, 255, 255); text-decoration: none;"
                  >
                    Apollo
                  </a>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="makeStyles-grow-1"
          />
          <div
            class="indicator"
          />
          <div
            style="margin-right: 10px;"
          >
            <button
              aria-controls="Lang-switcher"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
              data-testid="langBtn"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <div
                  style="font-size: 18px;"
                >
                  EN
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="makeStyles-sectionDesktop-8"
          >
            <button
              aria-label="show 4 new mails"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <span
                  class="MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                  >
                    4
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="show 17 new notifications"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <span
                  class="MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
                  >
                    17
                  </span>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="primary-search-account-menu"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="makeStyles-sectionMobile-9"
          >
            <button
              aria-controls="primary-search-account-menu-mobile"
              aria-haspopup="true"
              aria-label="show more"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div>
        content
      </div>
    </div>
  </div>
  <footer
    class="makeStyles-footer-12"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Footer
    </h6>
    <p
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Something here to give the footer a purpose!
    </p>
  </footer>
</DocumentFragment>
`;
